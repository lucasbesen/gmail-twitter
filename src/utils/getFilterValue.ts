import queryString from "query-string";

type GetFilterValueType = (locationSearch: string) => string;

const getFilterValue: GetFilterValueType = locationSearch => {
  const searchObj = queryString.parse(locationSearch);

  if (searchObj && searchObj.search) {
    const search = searchObj.search.toString();
    return search;
  }
  return '';
};

export default getFilterValue;
