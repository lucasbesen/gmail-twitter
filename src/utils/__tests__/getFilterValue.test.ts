import getFilterValue from '../getFilterValue';

it('should return yes on getFilterValue', () => {
  expect(getFilterValue('?search=yes')).toMatchSnapshot();
});
