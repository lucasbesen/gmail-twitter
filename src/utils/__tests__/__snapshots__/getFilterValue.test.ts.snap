// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return yes on getFilterValue 1`] = `"yes"`;
